### PR TITLE
Feature/luxonis dataset integration

### PR DESCRIPTION
**Pull request by sokovninn**
This PR includes:

- Added LuxonisDataset integration
- Added conversion to COCO and YOLO dataset formats (default is raw)
- Added script for dataset merging
- Added unique image names with uuid

### CODE REVIEW

File: datadreamer/utils/ldf_convreter.py
Comment (conorsim) [inline datadreamer/utils/ldf_convreter.py:1]: It seems like we have a duplicate file here with a typo

File: requirements.txt [hunk 1]
```diff
@@ -12,3 +12,4 @@ accelerate>=0.25.0
 scipy>=1.10.0
 bitsandbytes>=0.42.0
 nltk>=3.8.1
+luxonis-ml[all]
```
Comment (conorsim) [inline requirements.txt:15]: We could pin this to `luxonis-ml[all]>=0.1.0`

**Review by conorsim — COMMENTED** [review]
Looks good generally and I tried it out locally - thanks! 

I've left some comments on unimportant details. Maybe the most important being that there is this duplicate `ldf_convreter.py` file we likely want to delete.

File: datadreamer/pipelines/generate_dataset_from_scratch.py [hunk 2]
```diff
@@ -417,7 +417,9 @@ def main():
             prompts, prompt_objects
         ):
             for generated_image in generated_images_batch:
-                image_path = os.path.join(save_dir, f"image_{num_generated_images}.jpg")
+                unique_id = uuid.uuid4().hex
```
[the remark this answers is not in the document]
Comment (sokovninn) [reply datadreamer/pipelines/generate_dataset_from_scratch.py:420]: I added it to simplify the merging process and avoid conflicts between image names.

File: requirements.txt (hunk 1 above)
Comment (sokovninn) [reply requirements.txt]: Done

File: datadreamer/utils/ldf_convreter.py
Comment (sokovninn) [reply datadreamer/utils/ldf_convreter.py:1]: Fixed, thanks.

File: datadreamer/utils/dataset_utils.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            annotations[image_name]["boxes"] = bboxes.tolist()
+
+    annotations["class_names"] = class_names
+
+    # Save to JSON file
+    with open(os.path.join(save_dir, file_name), "w") as f:
+        json.dump(annotations, f, indent=4)
+
+
+def convert_to_ldf(
```
[the remark this answers is not in the document]
Comment (sokovninn) [reply datadreamer/utils/dataset_utils.py]: It's redundant, thanks.

File: datadreamer/utils/base_converter.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+import numpy as np
+
+
+class BaseConverter(ABC):
+    """Abstract base class for converter."""
+
+    @abstractmethod
+    def convert(self, dataset_dir, output_dir, split_ratios, copy_files=True):
+        """Converts a dataset into a format suitable for training with YOLO, including
```
[the remark this answers is not in the document]
Comment (sokovninn) [reply datadreamer/utils/base_converter.py]: Fixed it

**Review by HonzaCuhel — COMMENTED** [review]
LGTM, but wouldn't it be a good idea to write unit tests for the dataset conversion (to COCO, YOLO and LuxonisDataset format)? @sokovninn

**Comment by sokovninn** [comment]
@HonzaCuhel sure, I'll add some unit tests for converters.

**Pull request merged**